openapi/schemas/processes-dru/cwl.yaml

```diff
@@ -0,0 +1,1650 @@
+      description: The compute capability supported by the GPU hardware.
+      pattern: '^\d+\.\d+$'
+    - $ref: '#/$definitions/CUDAComputeCapabilityArray'
+    title: CUDA compute capability
+    description: |
+      The compute capability supported by the GPU hardware.
+      
+      * If this is a single value, it defines only the minimum compute capability.
+        GPUs with higher capability are also accepted.
+      * If it is an array value, then only select GPUs with compute capabilities that explicitly
+        appear in the array.
+        See https://docs.nvidia.com/deploy/cuda-compatibility/#faq and
+        https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#cuda-compute-capability
+        for details.
+  CUDAComputeCapabilityArray:
+    type: array
+    title: CUDAComputeCapabilityArray
+    items:
+      type: string
+      title: CUDA compute capability
+      description: The compute capability supported by the GPU hardware.
+      pattern: '^\d+\.\d+$'
+    minItems: 1
+  cwltool:CUDARequirement:
+    type: object
+    title: cwltool:CUDARequirement
+    properties:
+      cudaVersionMin:
+        type: string
+        title: CUDA version minimum
+        description: | 
+          The minimum CUDA version required to run the software. This corresponds to a CUDA SDK release.
+
+          When run in a container, the container image should provide the CUDA runtime,
+          and the host driver is injected into the container.  In this case, because CUDA drivers
+          are backwards compatible, it is possible to use an older SDK with a newer driver across major versions.
+
+          See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.
+        example: '11.4'
+        pattern: '^\d+\.\d+$'
+      cudaComputeCapability:
+        $ref: '#/$definitions/CUDAComputeCapability'
+      cudaDeviceCountMin:
+        type: integer
+        title: CUDA device count minimum
+        description: The minimum amount of devices required.
+        default: 1
+        example: 1
+        minimum: 1
+      cudaDeviceCountMax:
+        type: integer
+        title: CUDA device count maximum
+        description: The maximum amount of devices required.
+        default: 1
+        example: 8
+        minimum: 1
+    required:
+    - cudaVersionMin
+    - cudaComputeCapability
+    additionalProperties: {}
+  CUDARequirementMap:
+    type: object
+    title: CUDARequirementMap
+    properties:
+      cwltool:CUDARequirement:
+        $ref: '#/$definitions/cwltool:CUDARequirement'
+    required:
+    - cwltool:CUDARequirement
+    additionalProperties: {}
+  OGCAPIRequirement:
+    type: object
+    title: OGCAPIRequirement
+    description: | 
+      Hint indicating that the Application Package corresponds to an
+      OGC API - Processes provider that should be remotely executed and monitored
+      by this instance. (note: can only be an 'hint' as it is unofficial CWL specification).
```

Comment:
If the process is not otherwise executable without making use of the `OGCAPIRequirement` then it should be under `requirements` not `hints`. Non-standard extensions are allowed in both fields

Reply:
I agree. 
I think it is implementation dependent whether this would be required or optional on the target platform where the process is being deployed. However, I would like to make this definition proposal somewhat established such that different implementators could be guided toward a somewhat interoperable reference.